### PR TITLE
Issue #2703109 by scottrigby, laxmanghavte: Author field should be customizable

### PR DESCRIPTION
**Pull request by ghavtelaxman**
Please review

### CODE REVIEW

**Comment by scottrigby** [comment]
@laxman-faichi I wouldn't make Token a dependency, but optional. You could wrap the Token related changes in `module_exists()`, and add a description that says something like "keep blank to use the entity author, add a static string, or for dynamic strings enable the Token module". Thoughts?

File: modules/fb_instant_articles_display/fb_instant_articles_display.info [hunk 1]
```diff
@@ -3,5 +3,6 @@ description = Display module for Facebook Instant Articles.
 package = Facebook Instant Articles
 core = 7.x
 dependencies[] = ctools
+dependencies[] = token
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/fb_instant_articles_display.info:6]: Remove dependency

File: modules/fb_instant_articles_display/fb_instant_articles_display.module [hunk 2]
```diff
@@ -11,6 +11,50 @@
...
+
+  $form['fb_instant_articles_display']['author']['fb_instant_articles_display_author'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Facebook instant articles node author'),
+    '#description' => 'keep blank to use the entity author, add a static string, or for dynamic strings enable the Token module e.g [node:author] ' . $token_listing_link,
+    '#default_value' => variable_get('fb_instant_articles_display_author', '[node:author]'),
+    '#size' => 60,
+  );
+
+
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/fb_instant_articles_display.module:52]: Remove extra new lines. One should suffice.

File: modules/fb_instant_articles_display/src/EntityPropertyMapper.php [hunk 3]
```diff
@@ -109,16 +109,27 @@ private function addHeaderFromProperties() {
       );
     }
 
+    // user token or static string for author name
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/src/EntityPropertyMapper.php:112]: Capitalize comment and end with punctuation. See Drupal's [API documentation and comment standards](https://www.drupal.org/coding-standards/docs#drupal).

File: modules/fb_instant_articles_display/src/EntityPropertyMapper.php [hunk 4]
```diff
@@ -109,16 +109,27 @@ private function addHeaderFromProperties() {
...
+    $author = '';
+    if (module_exists('Token')) {
+      $display_author = variable_get('fb_instant_articles_display_author', '[node:author]');
+      $author = token_replace($author_token, array('node' => $node));
+    } else {
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/src/EntityPropertyMapper.php:117]: Move the `else` control to a new line. See Drupal's [Coding standards / Control Structures](https://www.drupal.org/coding-standards#controlstruct).

File: modules/fb_instant_articles_display/src/EntityPropertyMapper.php [hunk 5]
```diff
@@ -109,16 +109,27 @@ private function addHeaderFromProperties() {
       );
     }
 
+    // user token or static string for author name
+    $author = '';
+    if (module_exists('Token')) {
+      $display_author = variable_get('fb_instant_articles_display_author', '[node:author]');
+      $author = token_replace($author_token, array('node' => $node));
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/src/EntityPropertyMapper.php:116]: `$author_token` is not defined. I think you meant `$display_author` here. Like:

``` diff
-      $author = token_replace($author_token, array('node' => $node));
+      $author = token_replace($display_author, array('node' => $node));
```

File: modules/fb_instant_articles_display/src/EntityPropertyMapper.php [hunk 6]
```diff
@@ -109,16 +109,27 @@ private function addHeaderFromProperties() {
...
-        $header->addAuthor(
-          Author::create()
-            ->withName($author->name)
-        );
+      if (empty($author)) {
+        $display_author = user_load($this->entity->uid);
+        $author = $display_author->name;
       }
     }
+    if ($author) {
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/src/EntityPropertyMapper.php:127]: If you check `if (!empty($author)) {` you don't need to define the empty string `$author = '';` above on line 113 (cleaner not to do that if you don't need to).

File: modules/fb_instant_articles_display/src/EntityPropertyMapper.php [hunk 7]
```diff
@@ -109,16 +109,27 @@ private function addHeaderFromProperties() {
       );
     }
 
+    // user token or static string for author name
+    $author = '';
```
Comment (scottrigby) [inline modules/fb_instant_articles_display/src/EntityPropertyMapper.php:113]: No need to define this is you change the if statement below. See this comment below: https://github.com/BurdaMagazinOrg/module-fb_instant_articles/pull/71/files#r67538644

**Comment by scottrigby** [comment]
@laxman-faichi Other than those comments, this looks good

**Comment by m4olivei** [comment]
Closing as this hasn't seen any activity in awhile. Feel free to reopen if there is interest. I'm personally not maintaining the 7.x version, but happy to merge if RTBC.

**Pull request closed without merging**